node/src/components/consensus/era_supervisor/tests.rs

```diff
@@ -169,105 +219,196 @@ fn new_test_chainspec(stakes: Vec<(PublicKey, u64)>) -> Chainspec {
-    let mut effects = handle(&mut es, event);
-
-    // As a result, the era supervisor should request validation of the proto block and evidence
-    // against Alice.
+    .received(NodeId(1), era_id);
+    let mut effects = handle(es, event);
     let validate = tokio::spawn(effects.pop().unwrap());
-    let request_evidence = tokio::spawn(effects.pop().unwrap());
-    assert!(effects.is_empty());
-    reactor.expect_send_message(NodeId(1)).await; // Sending request for evidence.
-    request_evidence.await.unwrap();
-
-    // The block validator returns true: The deploys in the block are valid. As a result, we don't
-    // do anything yet (no new effects): We have to wait for the evidence first.
     reactor
         .expect_block_validation(&proto_block, NodeId(1), timestamp, true)
         .await;
     let rv_event = validate.await.unwrap().pop().unwrap();
-    let effects = handle(&mut es, rv_event);
+
+    // As a result, the era supervisor should request validation of the proto block and evidence
+    // against Alice.
+    for _ in &accusations {
+        let request_evidence = tokio::spawn(effects.pop().unwrap());
+        reactor.expect_send_message(NodeId(1)).await; // Sending request for evidence.
+        request_evidence.await.unwrap();
+    }
     assert!(effects.is_empty());
 
-    // Node 1 replies with evidence of Alice's fault. That completes our requirements for the
-    // proposed block. The era supervisor announces that it has passed the proposed block to the
-    // consensus protocol. It also gossips the new evidence to other nodes.
-    let event = ClMessage::Evidence(alice_pk).received(NodeId(1), EraId(0));
-    let mut effects = handle(&mut es, event);
-    let _announce_proposed = tokio::spawn(effects.pop().unwrap());
-    let broadcast_evidence = tokio::spawn(effects.pop().unwrap());
+    // Node 1 replies with requested evidence.
+    for pk in &accusations {
+        let event = ClMessage::Evidence(*pk).received(NodeId(1), era_id);
+        let mut effects = handle(es, event);
+        let broadcast_evidence = tokio::spawn(effects.pop().unwrap());
+        assert!(effects.is_empty());
+        reactor.expect_broadcast().await; //Gossip evidence to other nodes.
+        broadcast_evidence.await.unwrap();
+    }
+
+    // The block validator returns true: The deploys in the block are valid. That completes our
+    // requirements for the proposed block. The era supervisor announces that it has passed the
+    // proposed block to the consensus protocol.
+    let mut effects = handle(es, rv_event);
+    let announce_proposed = tokio::spawn(effects.pop().unwrap());
     assert!(effects.is_empty());
     assert_eq!(proto_block, reactor.expect_proposed().await);
-    // TODO: Why does it hang if we await this effect?
-    // announce_proposed.await.unwrap();
-    reactor.expect_broadcast().await; //Gossip evidence to other nodes.
-    broadcast_evidence.await.unwrap();
+    announce_proposed.await.unwrap();
```

Comment:
Proposal announcements are removed after #555 (you might find that out when trying to run tests on merged code :wink:).